### PR TITLE
Fixes-21021 Display proxy cache in project configuration page

### LINKED ISSUE #21021: Unable to view the speed limit for proxy cache project
I can set the speed limit when creating the proxy cache project, but I have nowhere to view and update it afterward.

It returned from the backend, and we should add it into the configuration page,

```
{
    "creation_time": "2024-10-10T07:13:26.723Z",
    "current_user_role_id": 1,
    "current_user_role_ids": [
        1
    ],
    "cve_allowlist": {
        "creation_time": "0001-01-01T00:00:00.000Z",
        "id": 2,
        "items": [],
        "project_id": 2,
        "update_time": "0001-01-01T00:00:00.000Z"
    },
    "metadata": {
        "proxy_speed_kb": "-1",
        "public": "false",
        "retention_id": "1"
    },
    "name": "proxy",
    "owner_id": 1,
    "owner_name": "admin",
    "project_id": 2,
    "registry_id": 1,
    "repo_count": 1,
    "update_time": "2024-10-10T07:13:26.723Z"
}
```

It should provide an way to the project admin to do this.

image

### PR DESCRIPTION
**Pull request by xuelichao**
Display proxy cache in project configuration page

# Issue being fixed
Fixes #21021

This PR Only supports to display the values:
Screenshot 2024-10-15 at 17 55 09

Please indicate you've done the following:
- [X] Well Written Title and Summary of the PR
- [X] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [X] Accepted the DCO. Commits without the DCO will delay acceptance.
- [X] Made sure tests are passing and test coverage is added if needed.
- [X] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/portal/src/app/base/project/project-config/project-policy-config/project.ts [hunk 1]
```diff
@@ -30,5 +32,6 @@ export class Project {
         this.metadata.severity = 'low';
         this.metadata.auto_scan = false;
         this.metadata.auto_sbom_generation = false;
+        this.metadata.bandwidth = -1;
```
Comment (bupd) [inline src/portal/src/app/base/project/project-config/project-policy-config/project.ts:35]:
```suggestion
        this.metadata.proxy_speed_kb = -1;
```

File: src/portal/src/app/base/project/project-config/project-policy-config/project.ts [hunk 2]
```diff
@@ -21,6 +22,7 @@ export class Project {
         auto_scan: string | boolean;
         auto_sbom_generation: string | boolean;
         reuse_sys_cve_allowlist?: string;
+        bandwidth?: number | null;
```
Comment (bupd) [inline src/portal/src/app/base/project/project-config/project-policy-config/project.ts:25]:
```suggestion
        proxy_speed_kb?: number | null;
```

File: src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.ts [hunk 3]
```diff
@@ -55,8 +67,12 @@ export class ProjectPolicy {
         }
         this.ScanImgOnPush = pro.metadata.auto_scan === 'true';
         this.GenerateSbomOnPush = pro.metadata.auto_sbom_generation === 'true';
+        this.ProxyCacheEnabled = pro.registry_id ? true : false;
+        this.RegistryId = pro.registry_id;
+        this.Bandwidth = pro.metadata.bandwidth ? pro.metadata.bandwidth : -1;
```
Comment (bupd) [inline src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.ts:72]:
```suggestion
        this.Bandwidth = pro.metadata.proxy_speed_kb
            ? pro.metadata.proxy_speed_kb
            : -1;
```

**Review by bupd — COMMENTED** [review]
Hello @xuelichao, I tested the changes on my local environment, and I noticed that the bandwidth is the same for all proxy cache projects, showing -1. I’ve made some changes to fix this so that the bandwidth is displayed correctly for each proxy cache project instead of -1 across the board.


fix this

File: src/portal/src/app/base/project/project-config/project-policy-config/project.ts (hunk 1 above)
Comment (xuelichao) [reply src/portal/src/app/base/project/project-config/project-policy-config/project.ts]: done

File: src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.ts (hunk 3 above)
Comment (xuelichao) [reply src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.ts]: done

**Review by stonezdj — APPROVED** [review]

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**